src/main/java/v/hryhoryk/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -1,10 +1,23 @@
 package v.hryhoryk.onlinebookstore.repository.book;
 
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 import v.hryhoryk.onlinebookstore.model.Book;
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    Optional<Book> findById(Long id);
+
+    @Query(value = "from Book b join fetch b.categories c where c.id = :id")
+    List<Book> findBooksByCategoryId(Long id);
+
+    @Query(value = "from Book b join fetch b.categories where b.isDeleted = false")
```

Comment:
the same how it works without query?

Reply:
Nope, without query it's not working at all